cmd/server/controllers/BuyerController.go

```diff
@@ -0,0 +1,127 @@
+package controllers
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/vinigracindo/mercado-fresco-stranger-strings/internal/domains/buyer"
+)
+
+type BuyerController struct {
+	service buyer.Service
+}
+
+func NewBuyer(service buyer.Service) BuyerController {
+	return BuyerController{service: service}
+}
+
+func (c *BuyerController) Store() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req request
+		if err := ctx.ShouldBindJSON(&req); err != nil {
+			ctx.AbortWithStatusJSON(http.StatusUnprocessableEntity, gin.H{
+				"error":   err.Error(),
+				"message": "Campos obrigatórios não preenchidos"})
+			return
+		}
+
+		buyer, err := c.service.Store(req.Id, req.CardNumberId, req.FirstName, req.LastName)
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
```

Comment:
No caso do create nao precisamos retornar 404 (not found), o ideal  é passar o 409 (conflict), por já existir.